// @quansync/fs@7.3.2/dist/index.js
import fs from "node:fs";
import { quansync } from "quansync";

//#region src/index.ts
const readFile = quansync({
	sync: fs.readFileSync,
	async: fs.promises.readFile
});
const writeFile = quansync({
	sync: fs.writeFileSync,
	async: fs.promises.writeFile
});
const unlink = quansync({
	sync: fs.unlinkSync,
	async: fs.promises.unlink
});
const access = quansync({
	sync: fs.accessSync,
	async: fs.promises.access
});
const stat = quansync({
	sync: fs.statSync,
	async: fs.promises.stat
});
const lstat = quansync({
	sync: fs.lstatSync,
	async: fs.promises.lstat
});
const cp = quansync({
	sync: fs.copyFileSync,
	async: fs.promises.copyFile
});
const rm = quansync({
	sync: fs.rmSync,
	async: fs.promises.rm
});
const mkdir = quansync({
	sync: fs.mkdirSync,
	async: fs.promises.mkdir
});
const rename = quansync({
	sync: fs.renameSync,
	async: fs.promises.rename
});
const readdir = quansync({
	sync: fs.readdirSync,
	async: fs.promises.readdir
});
const realpath = quansync({
	sync: fs.realpathSync,
	async: fs.promises.realpath
});
const readlink = quansync({
	sync: fs.readlinkSync,
	async: fs.promises.readlink
});
const symlink = quansync({
	sync: fs.symlinkSync,
	async: fs.promises.symlink
});
const chown = quansync({
	sync: fs.chownSync,
	async: fs.promises.chown
});
const lchown = quansync({
	sync: fs.lchownSync,
	async: fs.promises.lchown
});
const chmod = quansync({
	sync: fs.chmodSync,
	async: fs.promises.chmod
});
const utimes = quansync({
	sync: fs.utimesSync,
	async: fs.promises.utimes
});
const lutimes = quansync({
	sync: fs.lutimesSync,
	async: fs.promises.lutimes
});
const mkdtemp = quansync({
	sync: fs.mkdtempSync,
	async: fs.promises.mkdtemp
});

//#endregion
export { access, chmod, chown, cp, lchown, lstat, lutimes, mkdir, mkdtemp, readFile, readdir, readlink, realpath, rename, rm, stat, symlink, unlink, utimes, writeFile };
